db/sequence_allocator_test.go

```diff
@@ -119,6 +120,61 @@ func TestReleaseSequencesOnStop(t *testing.T) {
 
 }
 
+// Reproduces deadlock from CBG-663.  Required adding a sleep inside the <-time.After case in
+// releaseSequenceMonitor to reliably queue up a large number of sequence allocation requests
+// between <-time.After fires and releaseUnusedSequences is called (where previously reserveNotify would block)
+func TestSequenceAllocatorDeadlock(t *testing.T) {
+
+	t.Skip("Requires additional sleep in production code to reliably hit race")
+
+	var a *sequenceAllocator
+	var err error
+
+	var wg sync.WaitGroup
+	callbackCount := 0
+	incrCallback := func() {
+		callbackCount++
+		if callbackCount == 2 {
+			// queue up a number of sequence requests
+			// Wait for 500ms for releaseSequenceMonitor time.After to trigger
+			time.Sleep(100 * time.Millisecond)
+
+			for i := 0; i < 500; i++ {
+				wg.Add(1)
+				go func(a *sequenceAllocator) {
+					a.nextSequence()
```

Comment:
Just adding a test assertion for this error.

```suggestion
					_, err := a.nextSequence()
					assert.NoError(t, err)
```